doc/faq.md

```diff
@@ -57,6 +57,65 @@ directory. None of this should affect any existing Haskell tools at all.
   there may be more options here for new projects in the future (see issue
   [253](https://github.com/commercialhaskell/stack/issues/253))
 
+### Cabal-the-library
+
+The relationship between stack and Cabal-the-library is complicated.
+
+1. Stack itself builds against a version of the Cabal library, which it uses for parsing Cabal files.
+
+2. Stack wraps functionality from a version of the Cabal library in order to build packages.
+
+    i. For packages with `build-type: Simple`, Stack uses a setup executable compiled with a version of the Cabal library to perform builds. All packages using the same compiler and Cabal version are built with the same executable. These executables are cached in the `setup-exe-cache` configuration directory.
+
+    ii. For packages with `build-type: Custom`, Stack compiles `Setup.hs` against a version of the Cabal library and uses that setup executable to perform builds. The executable is stored TODO
+
+3. Packages themselves may depend on the Cabal library.
+
+There are three (nearly) corresponding versions of the Cabal library which are relevant to a build.
+
+1. The version used to compile stack, which is not necessarily present on stack users’ machines.
+
+   Stack will always be compiled using the most recent stable version of Cabal-the-library.
```

Comment:
Not really true. We may attempt to do that, but we could for example release a point release of an older Stack release without upgrading to the newest Cabal library version.

Reply:
Added 'where possible' and made clear this refers to releases.

https://github.com/dbaynard/stack/blob/9900d99114967698223f43c84e4cedecc24cea97/doc/faq.md#L70-L72